File: src/main/java/com/polykhel/billnext/service/dto/WalletDTO.java
package com.polykhel.billnext.service.dto;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Objects;
import javax.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

/**
 * A DTO for the {@link com.polykhel.billnext.domain.Wallet} entity.
 */
@Getter
@Setter
public class WalletDTO implements Serializable {

    private Long id;

    @NotNull
    private String name;

    @NotNull
    private BigDecimal amount;

    private String currency;

    private String remarks;

    private WalletGroupDTO walletGroup;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof WalletDTO)) {
            return false;
        }

        WalletDTO walletDTO = (WalletDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, walletDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "WalletDTO{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", amount=" + getAmount() +
            ", currency='" + getCurrency() + "'" +
            ", remarks='" + getRemarks() + "'" +
            ", walletGroup=" + getWalletGroup() +
            "}";
    }
}

File: src/main/kotlin/com/polykhel/billnext/service/dto/WalletDTO.kt
package com.polykhel.billnext.service.dto

import java.io.Serializable
import java.math.BigDecimal
import java.util.*
import javax.validation.constraints.NotNull

/**
 * A DTO for the [com.polykhel.billnext.domain.Wallet] entity.
 */
data class WalletDTO(

    var id: Long? = null,

    @get: NotNull
    var name: String? = null,

    @get: NotNull
    var amount: BigDecimal? = null,

    var currency: String? = null,

    var remarks: String? = null,

    var walletGroup: WalletGroupDTO? = null
) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is WalletDTO) return false
        val walletDTO = other
        if (this.id == null) {
            return false
        }
        return Objects.equals(this.id, walletDTO.id)
    }

    override fun hashCode() = Objects.hash(this.id)
}
